TeachingRecordSystem/src/TeachingRecordSystem.Core/Jobs/PopulateAllPersonsSearchAttributesJob.cs

```diff
@@ -0,0 +1,18 @@
+using Hangfire;
+using Microsoft.EntityFrameworkCore;
+using TeachingRecordSystem.Core.DataStore.Postgres;
+using TeachingRecordSystem.Core.Jobs.Scheduling;
+
+namespace TeachingRecordSystem.Core.Jobs;
+
+[AutomaticRetry(Attempts = 0)]
+public class PopulateAllPersonsSearchAttributesJob(TrsDbContext dbContext, IBackgroundJobScheduler backgroundJobScheduler)
+{
+    public async Task Execute()
+    {
+        foreach (var person in dbContext.Persons.AsNoTracking())
```

Comment:
I'm not sure exactly how this works internally but I suspect this may not work with the data volume we have on production. It may be better to get an `IAsyncEnumerable` and also only select the attribute(s) we care about; we certainly don't want to be pulling every `Person` record into memory at one time.
```
await foreach (var personId in dbContext.Persons.Select(p => p.PersonId).AsAsyncEnumerable())
{
    await backgroundJobScheduler.Enqueue<PopulatePersonSearchAttributesJob>(j => j.Execute(personId));
}
```

Reply:
Funnily enough I did have it as IAsyncEnumerable originally but didn't think it would make it any difference to how it would buffer the results in memory i.e. as long as not adding ToListAsync() then it should NOT hold all records in memory even without IAsyncEnumerable

Reply:
If that's the case then existing enumerable is ok. I still think it's worth the `Select(p => p.PersonId)`, though.